### PR TITLE
Fix mod_final_25519

### PR DESCRIPTION
**Pull request by ppopth**

### CODE REVIEW

File: src/crypto/x25519/mod.rs [hunk 1]
```diff
@@ -549,19 +549,38 @@ fn mod_25519(x: Felem2) -> Felem {
 }
 
 fn mod_final_25519(x: Felem) -> Felem {
+    const MASK_64BITS: u128 = 0xffff_ffff_ffff_ffff;
```
Comment (Bren2010) [inline src/crypto/x25519/mod.rs:552]: Defining this as a const at the top and replacing all instances would definitely be welcome, along with `0x7fff_...`.

File: src/crypto/x25519/mod.rs [hunk 2]
```diff
@@ -549,19 +549,38 @@ fn mod_25519(x: Felem2) -> Felem {
...
+    top = acc1.wrapping_add(top) >> 64;
+    top = acc2.wrapping_add(top) >> 64;
+    top = acc3.wrapping_add(top) >> 63;
     top = top.wrapping_mul(19);
+
     acc0 = acc0.wrapping_add(top);
     acc1 = acc1.wrapping_add(acc0 >> 64);
     acc2 = acc2.wrapping_add(acc1 >> 64);
     acc3 = acc3.wrapping_add(acc2 >> 64);
+    acc3 &= 0x7fff_ffff_ffff_ffff;
```
Comment (Bren2010) [inline src/crypto/x25519/mod.rs:583]: What is this extra code doing? It doesn't make sense to me

File: src/crypto/x25519/mod.rs (hunk 1 above)
Comment (vkrasnov) [reply src/crypto/x25519/mod.rs:552]: Good idea.

File: src/crypto/x25519/mod.rs (hunk 2 above)
Comment (vkrasnov) [reply src/crypto/x25519/mod.rs:583]: Looks like masking the top bit of the output. However is it not sufficient to execute just a single extra round of mul + add?

Comment (ppopth) [reply src/crypto/x25519/mod.rs]: I put the comments in the source code. What do you think?

**Review by vkrasnov — APPROVED** [review]

**Review by Bren2010 — CHANGES_REQUESTED** [review]

**Review by Bren2010 — APPROVED** [review]

**Pull request merged**